dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnectorTest.java

```diff
@@ -44,4 +50,52 @@ public void buildProperties_success() {
     assertEquals(EXAMPLE_PASSWORD, actual.getProperty("password"));
     assertEquals(EXAMPLE_USER, actual.getProperty("user"));
   }
+
+  @Test
+  public void buildUrl_providedBoth_throwsException() {
+    when(arguments.getOracleServicename()).thenReturn("ORCLPDB");
+    when(arguments.getOracleSID()).thenReturn("ORCLPDB1");
+    // Act
+    ThrowingRunnable runnable = () -> AbstractOracleConnector.buildUrl(arguments);
+    // Assert
+    assertThrows(MetadataDumperUsageException.class, runnable);
+  }
+
+  @Test
+  public void buildUrl_providedNone_throwsException() {
+    ThrowingRunnable runnable = () -> AbstractOracleConnector.buildUrl(arguments);
+    assertThrows(MetadataDumperUsageException.class, runnable);
```

Comment:
I would suggest asserting the error message of the exception:
1. to make sure that the exception happened for the expected reason,
2. to check that the correct message is displayed to the user in this case.

Reply:
Done, I added a method for creating this exception and verification that they match by the message.